Sources/PenguinStructures/Lens.swift

```diff
@@ -0,0 +1,42 @@
+//******************************************************************************
+// Copyright 2020 Penguin Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/// `KeyPath`s with a statically known `Value` endpoint.
+// This protocol is only needed because we can't represent `: KeyPath<_,V>` as a constraint in the
```

Comment:
Nit: the jazzy documentation system doesn't seem to successfully handle non-doc comments between the doc comments and the element the doc comment annotates. I recommend either hoisting this above the doc comment, or making this part of the doc comment. (I think in this case, the latter option is preferred.)